src/iexamine.cpp

```diff
@@ -1488,17 +1495,26 @@ void iexamine::chainfence( Character &you, const tripoint &examp )
         return;
     }
 
-    map &here = get_map();
     int move_cost = 400;
     // TODO: Remove hardcoded trait checks when new arthropod bits happen
     if( here.has_flag( ter_furn_flag::TFLAG_CLIMB_SIMPLE, examp ) ) {
 
         if( you.has_proficiency( proficiency_prof_parkour ) ) {
-            add_msg( _( "You vault over the obstacle with ease." ) );
-            move_cost = 100; // Not tall enough to warrant spider-climbing, so only relevant trait.
+            if( you.has_trait( trait_BADKNEES ) ) {
+                add_msg( _( "Your freerunning ability mostly makes up for your bad knees on the way over this obstacle." ) );
+                move_cost = 150;
+            } else {
+                add_msg( _( "You vault over the obstacle with ease." ) );
+                move_cost = 100; // Not tall enough to warrant spider-climbing, so only relevant trait.
+            }
         } else {
-            add_msg( _( "You vault over the obstacle." ) );
-            move_cost = 300; // Most common move cost for barricades pre-change.
+            if( you.has_trait( trait_BADKNEES ) ) {
+                add_msg( _( "Your bad knees slow you down on the way over this obstacle." ) );
+                move_cost = 450;
```

Comment:
Feels like this should use movecost_obstacle_modifier from the mutation and a generic faster/slower than normal message rather than try to invent new numbers or do per-mutation tree of messages ( or equivalent enchantment/limb score if one exists )

Reply:
There's already a per-mutation tree of messages here, and more in slip_down. I like the flavor, and even added some to the slip_down in a recent PR.

Reply:
I agree that movecost_obstacle_modifier would be more consistent and repeatable. I think it would change the existing behaviors here if we went that way, though.